### PR TITLE
Auto create dirs

### PR DESCRIPTION
**Pull request by lrcowton**
Various changes as discussed. Readme has been updated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,43 @@
-# Parallel
-Run functions in parallel using Faculty Jobs
+# faculty-parallel
```
Comment (acroz) [inline README.md]:
```suggestion
# faculty-distributed
```

File: faculty_distributed/__init__.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        else:
+            self.tmpdir = tempfile.mkdtemp(prefix=self.tmpdir_prefix)
+
+        for jobs_dir in [
+            os.path.join(self.tmpdir, "output"),
+            os.path.join(self.tmpdir, "func"),
+        ]:
+            os.makedirs(jobs_dir)
+
+    def map(self, func, args_sequence):
```
Comment (pbugnion) [inline faculty_distributed/__init__.py]: It's fairly customary to group public methods either at the top (below the constructor) or at the bottom.

File: faculty_distributed/__init__.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        """
+        Execute function for each set of arguments in list in parallel using
+        Faculty Jobs.
```
Comment (pbugnion) [inline faculty_distributed/__init__.py]: Probably worth noting `kwargs` are unsupported.

File: faculty_distributed/__init__.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        ----------
+        func: function object
+            generic python function
+        args_sequence: list
+            list of lists of arguments to be iterated over in parallel
+
+        Returns
+        -------
+        output: list
+            list of outputs of function from each job
```
Comment (pbugnion) [inline faculty_distributed/__init__.py]: An example usage here might be good?

**Comment by pbugnion** [comment]
I think this is great.

I'd make one comment, possibly for future reference. You (and most people in the consultancy) put a lot of effort into documenting private methods.

While adding documentation is laudable, too much documentation makes code hard to change. In particular, it isn't obvious to me that adding docstrings to private methods always warrants the additional maintenance cost incurred.

I don't suggest changing this now, but it is worth bearing in mind for the future.

**Pull request merged**